### PR TITLE
`#[deny(unsafe_op_in_unsafe_fn)]` in sys/cloudabi

### PR DESCRIPTION
**Pull request by chansuke**
Partial fix of #73904.

This encloses unsafe operations in unsafe fn in sys/cloudabi.

### CODE REVIEW

**Comment by KodrAus** [comment]
Thanks @chansuke!

@bors r+

File: library/std/src/sys/cloudabi/abi/cloudabi.rs [hunk 1]
```diff
@@ -2402,7 +2405,7 @@ pub unsafe fn file_readdir(
     cookie_: dircookie,
     bufused_: &mut usize,
 ) -> errno {
-    cloudabi_sys_file_readdir(fd_, buf_.as_mut_ptr() as *mut (), buf_.len(), cookie_, bufused_)
+    unsafe fn cloudabi_sys_file_readdir(fd_, buf_.as_mut_ptr() as *mut (), buf_.len(), cookie_, bufused_) }
```
Comment (JohnTitor) [inline library/std/src/sys/cloudabi/abi/cloudabi.rs:2408]:
```suggestion
    unsafe { cloudabi_sys_file_readdir(fd_, buf_.as_mut_ptr() as *mut (), buf_.len(), cookie_, bufused_) }
```

Comment (chansuke) [reply library/std/src/sys/cloudabi/abi/cloudabi.rs:2408]: Thank you for your review.

**Review by JohnTitor — APPROVED** [review]
I think this should be fine now. @KodrAus could you take another look?

**Comment by KodrAus** [comment]
Alrighty, let's try run this one through again.

@bors r+

**Comment by crlf0710** [comment]
@chansuke Ping from triage, it seems there's a compilation error. Since this platform is not covered by default in pr builds, consider test building it locally. (or temporarily tweak ci configuration to test it.) Thanks!

**Comment by chansuke** [comment]
@crlf0710 Thanks, I'll take a look at the compilation error.

**Comment by crlf0710** [comment]
@chansuke  Ping from triage, any updates on this?

**Comment by chansuke** [comment]
@crlf0710 Sorry, no progress on this issue...

**Comment by chansuke** [comment]
@crlf0710 I fixed the compilation error, thanks.

**Pull request merged**